elfloader-tool/src/arch-riscv/boot.c

```diff
@@ -28,14 +35,28 @@
 // page table entry (PTE) field
 #define PTE_V     0x001 // Valid
 
-#define PTE64_PPN0_SHIFT 10
+#define PTE_PPN0_SHIFT 10
 
-#define PTES_PER_PT (RISCV_PGSIZE/sizeof(long))
+#if __riscv_xlen == 32
```

Comment:
It would be *nice* if these constants came from a include/arch-riscv/<32/64>/mode header, as we do for arm. I'm happy to fix this internally though after. 

For context we have been moving away from #ifdef based definitions, as we've found file includes to be less error prone and more reliable.

Reply:
@pingerino that is really good to know for future work. Feel free to make that change afterwards, and I'll make a note of it for future work.